packages/create-cloudflare/src/cli.ts

```diff
@@ -83,43 +83,100 @@ export const runCli = async (args: Partial<C3Args>) => {
 	});
 
-	// If not specified, attempt to infer the `type` argument from other flags
-	if (!args.type) {
-		if (args.framework) {
-			args.type = "webFramework";
-		} else if (args.existingScript) {
-			args.type = "pre-existing";
-		}
+	const validatedArgs: C3Args = {
+		...args,
+		projectName,
+	};
+
+	const originalCWD = process.cwd();
+	const { name, path } = setupProjectDirectory(validatedArgs);
+
+	const template = await selectTemplate(args);
+	const ctx: C3Context = {
+		project: { name, path },
+		args: validatedArgs,
+		template,
+		originalCWD,
+		gitRepoAlreadyExisted: await isInsideGitRepo(dirname(path)),
+		deployment: {
+			queues: {},
+			kvNamespaces: {},
+		},
+	};
+
+	await runTemplate(ctx);
+};
+
+const runTemplate = async (ctx: C3Context) => {
+	await create(ctx);
+	await configure(ctx);
+	await deploy(ctx);
+
+	await printSummary(ctx);
+};
+
+const create = async (ctx: C3Context) => {
+	const { template } = ctx;
+
+	if (template.generate) {
+		await template.generate(ctx);
 	}
 
-	const templateOptions = Object.entries(templateMap).map(
-		([value, { label, hidden }]) => ({ value, label, hidden })
-	);
+	await copyTemplateFiles(ctx);
 
-	const type = await processArgument<string>(args, "type", {
-		type: "select",
-		question: "What type of application do you want to create?",
-		label: "type",
-		options: templateOptions,
-		defaultValue: C3_DEFAULTS.type,
-	});
+	chdir(ctx.project.path);
+	await npmInstall(ctx);
+	await rectifyPmMismatch(ctx);
+
+	endSection(`Application created`);
+};
+
+const configure = async (ctx: C3Context) => {
+	startSection("Configuring your application for Cloudflare", "Step 2 of 3");
+
+	await installWrangler();
+	await installWorkersTypes(ctx);
+
+	await updatePackageJson(ctx);
+	await updateWranglerToml(ctx);
 
-	if (!type) {
-		crash("An application type must be specified to continue.");
+	const { template } = ctx;
+	if (template.configure) {
+		await template.configure({ ...ctx });
 	}
 
-	if (!Object.keys(templateMap).includes(type)) {
-		crash(`Unknown application type provided: ${type}.`);
+	await offerGit(ctx);
+	await gitCommit(ctx);
+
+	endSection(`Application configured`);
+};
+
+const deploy = async (ctx: C3Context) => {
+	await offerToDeploy(ctx);
+	await createProject(ctx);
+	await runDeploy(ctx);
+};
+
+// Detects if a newer version of c3 is available by comparing the version
+// specified in package.json with the `latest` tag from npm
+const isUpdateAvailable = async () => {
+	if (process.env.VITEST || process.env.CI || !isInteractive()) {
+		return false;
 	}
 
-	const validatedArgs: C3Args = {
-		...args,
-		type,
-		projectName,
-	};
+	// Use a spinner when running this check since it may take some time
+	const s = spinner(spinnerFrames.vertical, blue);
+	s.start("Checking if a newer version is available");
+	const latestVersion = await runCommand(
+		["npm", "info", "create-cloudflare@latest", "dist-tags.latest"],
+		{ silent: true, useSpinner: false }
+	);
```

Comment:
Probably not for this PR as it would be out of scope, but what if instead of running `npm info` we were to directly fetch from the npm registry? (as suggested here: https://github.com/cloudflare/workers-sdk/pull/4627#issuecomment-1862979270 and as we did for the workerd compat date)

Reply:
(Related: #4729)

Reply:
I'd then suggest to leave this as is here and amend it in #4729